syntax = "proto3";

package sales;

service SalesService {
    // Client-streaming: upload CSV file chunk by chunk
    rpc UploadCSV(stream UploadChunk) returns (UploadResponse);
    
    // Check job status
    rpc GetJobStatus(JobStatusRequest) returns (JobStatusResponse);
}

message UploadChunk {
    bytes data = 1;
    string filename = 2;  // optional, sent in first chunk
    string auth_token = 3;  // optional authentication token
}

message UploadResponse {
    string job_id = 1;
    string status = 2;  // processing, completed, error
    string message = 3;
    string download_url = 4;  // relative path or full URL
    ProcessingMetrics metrics = 5;  // processing metrics
}

message JobStatusRequest {
    string job_id = 1;
    string auth_token = 2;  // optional authentication token
}

message JobStatusResponse {
    string job_id = 1;
    string status = 2;
    string download_url = 3;
    string error_message = 4;
    ProcessingMetrics metrics = 5;  // processing metrics
}

message ProcessingMetrics {
    int64 processing_time_ms = 1;  // processing time in milliseconds
    int64 rows_processed = 2;  // number of rows successfully processed
    int64 rows_skipped = 3;  // number of rows skipped due to errors
    int64 departments_count = 4;  // number of unique departments
    int64 peak_memory_mb = 5;  // peak memory usage in MB (optional)
}

